internal/distro/rhel85/partition_tables.go

```diff
@@ -5,230 +5,317 @@ import (
-						Mountpoint:   "/boot",
-						FSTabOptions: "defaults",
-						FSTabFreq:    0,
-						FSTabPassNo:  0,
-					},
-				},
-				{
-					Start: 1460224,
-					Size:  imageOptions.Size/sectorSize - 1460224 - 100,
-					Type:  "0FC63DAF-8483-4772-8E79-3D69D8477DE4",
-					UUID:  "6264D520-3FB9-423F-8AB8-7A0A8E3D3562",
-					Filesystem: &disk.Filesystem{
-						Type:         "xfs",
-						UUID:         uuid.Must(newRandomUUIDFromReader(rng)).String(),
-						Label:        "root",
-						Mountpoint:   "/",
-						FSTabOptions: "defaults",
-						FSTabFreq:    0,
-						FSTabPassNo:  0,
-					},
-				},
+	}
+	return disk.Partition{
+		Start: start,
+		Size:  uint64(m.MinSize),
+		Filesystem: &disk.Filesystem{
+			Type:         "xfs",
+			UUID:         diskUUID,
+			Mountpoint:   m.Mountpoint,
+			FSTabOptions: "defaults",
+			FSTabFreq:    0,
+			FSTabPassNo:  0,
+		},
+	}
+}
+
+func createRootPartition(start uint64, size uint64, archName string, uuid string) disk.Partition {
+	if archName == "x86_64" || archName == "aarch64" {
+		return disk.Partition{
+			Start: start,
+			Size:  size,
+			Type:  "0FC63DAF-8483-4772-8E79-3D69D8477DE4",
+			UUID:  "6264D520-3FB9-423F-8AB8-7A0A8E3D3562",
+			Filesystem: &disk.Filesystem{
+				Type:         "xfs",
+				UUID:         uuid,
+				Label:        "root",
+				Mountpoint:   "/",
+				FSTabOptions: "defaults",
+				FSTabFreq:    0,
+				FSTabPassNo:  0,
+			},
+		}
+	}
+	if archName == "ppc64le" {
+		return disk.Partition{
+			Start: start,
+			Size:  size,
+			Filesystem: &disk.Filesystem{
+				Type:         "xfs",
+				UUID:         uuid,
+				Mountpoint:   "/",
+				FSTabOptions: "defaults",
+				FSTabFreq:    0,
+				FSTabPassNo:  0,
 			},
 		}
 	}
+	return disk.Partition{
+		Start:    start,
+		Size:     size,
+		Bootable: true,
+		Filesystem: &disk.Filesystem{
+			Type:         "xfs",
+			UUID:         uuid,
+			Mountpoint:   "/",
+			FSTabOptions: "defaults",
+			FSTabFreq:    0,
+			FSTabPassNo:  0,
+		},
+	}
+}
 
-	panic("unsupported EC2 arch: " + arch.Name())
+func createBootPartition(mountpoint string, start uint64, size uint64) disk.Partition {
+	if mountpoint == "/boot/efi" {
+		return disk.Partition{
+			Start: start,
+			Size:  size,
+			Type:  "C12A7328-F81F-11D2-BA4B-00A0C93EC93B",
+			UUID:  "68B2905B-DF3E-4FB3-80FA-49D1E773AA33",
+			Filesystem: &disk.Filesystem{
+				Type:         "vfat",
+				UUID:         "7B77-95E7",
+				Mountpoint:   "/boot/efi",
+				FSTabOptions: "defaults,uid=0,gid=0,umask=077,shortname=winnt",
+				FSTabFreq:    0,
+				FSTabPassNo:  2,
+			},
+		}
+	}
+	return disk.Partition{
+		Start: start,
+		Size:  size,
+		Type:  "C12A7328-F81F-11D2-BA4B-00A0C93EC93B",
+		UUID:  "68B2905B-DF3E-4FB3-80FA-49D1E773AA33",
+		Filesystem: &disk.Filesystem{
+			Type:         "xfs",
+			UUID:         "2AE383D1-F57C-4F04-A1BD-32FC12A578EA",
+			Mountpoint:   "/boot",
+			FSTabOptions: "defaults",
+			FSTabFreq:    0,
+			FSTabPassNo:  0,
+		},
```

Comment:
Might be cleaner if the common parts were set at first and then the three values that differ are set conditionally.  It would make it easier to identify how the two cases differ when following the code.

Reply:
Oh yeah this is a good point. Thanks for that!